### PR TITLE
Added bounding box function for polygons.

### PR DESCRIPTION
**Pull request by rubenvanstaden**
Adding boundary box function for Polygon object. This is needed to add a Label to the centre of a symmetric polygon (etc, via). I also updated the viewer to plot all labels with the same key. The current version overwrites the dict value, since the key is the same.

### CODE REVIEW

File: gdspy/__init__.py [hunk 1]
```diff
@@ -257,6 +257,25 @@ def area(self, by_spec=False):
         else:
             return 0.5 * abs(poly_area)
 
+    def get_bounding_box(self):
+        """
+        Returns the bounding box for this cell.
```
Comment (heitzmann) [inline gdspy/__init__.py]: This docstring should say "Return the bounding box of the polygon."

File: gdspy/__init__.py [hunk 2]
```diff
@@ -257,6 +257,25 @@ def area(self, by_spec=False):
...
+
+        Returns
+        -------
+        out : Numpy array[2,2] or ``None``
+            Bounding box of this cell [[x_min, y_min], [x_max, y_max]], or
+            ``None`` if the cell is empty.
```
Comment (heitzmann) [inline gdspy/__init__.py]: There is never a reason to return `None`

File: gdspy/__init__.py [hunk 3]
```diff
@@ -257,6 +257,25 @@ def area(self, by_spec=False):
...
+        """
+        bb = numpy.array(((1e300, 1e300), (-1e300, -1e300)))
```
Comment (heitzmann) [inline gdspy/__init__.py]: Start with `if len(self.points) == 0: return None`, then the code follows without the if-clause

**Comment by rubenvanstaden** [comment]
@heitzmann Sorry for my noob mistakes. This is my first open source contribution. The main purpose of adding the hash to Labels is to have all the labels printed to the Layout, given that I have a list of labels with the same key (64, 0). The current version only prints one label string to the Viewer. So the hash is used for comparison.

I am programmatically labeling different cells, before flattening the whole top-level cell, but the labels has to have a gdsnumber of 64. And incrementing the datatype for each label is extremely redundant. Specially for larger layouts.

I only used the first two parameters in the hash, since I assume there will never be more that one label with the same position, therefore it will be unique. 

I am basically doing an LVS extraction for Superconducting Electronics.

**Comment by heitzmann** [comment]
@rubenvanstaden I see what you mean. This is in fact a bug... I just went ahead and added your changes (https://github.com/heitzmann/gdspy/commit/5fd94974ba8a4171fdef3d2a22bea7d16727898e) without the need for a `defaultdict` or a hash.
Could you please review the changes in the bounding box code and open a separate PR for it? I'll close this PR because it mixes the 2 issues.

**Pull request closed without merging**